bson/src/test/unit/org/bson/RawBsonDocumentSpecification.groovy

```diff
@@ -161,6 +161,18 @@ class RawBsonDocumentSpecification extends Specification {
         rawDocument << createRawDocumentVariants()
     }
 
+    def 'should return RawBsonDocument for sub documents'() {
+        expect:
+        rawDocument.get('a') instanceof BsonInt32
+        rawDocument.get('b') instanceof BsonInt32
+        rawDocument.get('c') instanceof RawBsonDocument
```

Comment:
Expand this to also iterate/get fields of 'c', as well as checking the length of its ByteBuffer.  Otherwise it won't catch if the RawBsonDocument was constructed improperly.